classes/Http/Action/Talk/CreateProcessAction.php

```diff
@@ -88,40 +97,27 @@ public function __construct(
         $this->applicationEmail   = $applicationEmail;
         $this->applicationTitle   = $applicationTitle;
         $this->applicationEndDate = $applicationEndDate;
-
-        parent::__construct($twig, $urlGenerator);
-    }
-
-    /**
-     * @param $requestData
-     *
-     * @return TalkForm
-     */
-    private function getTalkForm($requestData): TalkForm
-    {
-        return new TalkForm($requestData, $this->purifier, [
-            'categories' => $this->talkHelper->getTalkCategories(),
-            'levels'     => $this->talkHelper->getTalkLevels(),
-            'types'      => $this->talkHelper->getTalkTypes(),
-        ]);
+        $this->twig               = $twig;
+        $this->urlGenerator       = $urlGenerator;
     }
 
-    public function processCreateAction(Request $request): Response
+    public function __invoke(HttpFoundation\Request $request): HttpFoundation\Response
     {
-        // You can only create talks while the CfP is open
         if (!$this->callForPapers->isOpen()) {
             $request->getSession()->set('flash', [
                 'type'  => 'error',
                 'short' => 'Error',
                 'ext'   => 'You cannot create talks once the call for papers has ended',
             ]);
 
-            return $this->redirectTo('dashboard');
+            $url = $this->urlGenerator->generate('dashboard');
+
+            return new HttpFoundation\RedirectResponse($url);
         }
 
         $user = $this->authentication->user();
 
-        $form = $this->getTalkForm([
+        $form = $this->createTalkForm([
```

Comment:
As the form is created here, we can't write unit tests that go further than the condition above.